internal/domain/power-status.go

```diff
@@ -18,16 +19,707 @@
+			updateHWState(xname,pcsmodel.PowerStateFilter_Undefined,
+				pcsmodel.ManagementStateFilter_available,
+				"Unable to read response body")
+			continue
+		}
+
+		switch (xnametypes.HMSType(ctype)) {
+			case xnametypes.NodeBMC:    fallthrough
+			case xnametypes.RouterBMC:  fallthrough
+			case xnametypes.ChassisBMC:
+				//Any valid response means ON
+				updateHWState(xname,pcsmodel.PowerStateFilter_On,
+					pcsmodel.ManagementStateFilter_available,"")
+
+			case xnametypes.Node:
+				//Nodes: look for "PowerState" in response payload.
+				var info rf.ComputerSystem
+				err = json.Unmarshal(stsBody, &info)
+				if (err != nil) {
+					glogger.Errorf("ERROR unmarshalling power payload for '%s': %v",
+						rqURL,err)
+					//Power state unknown, but got a response, so mgmt state OK
+					updateHWState(xname,pcsmodel.PowerStateFilter_Undefined,
+						pcsmodel.ManagementStateFilter_available,
+						"Unable to unmarshal power payload")
+					break
+				}
+				powerState,_ = pcsmodel.ToPowerStateFilter(info.PowerState)
+				updateHWState(xname,powerState,pcsmodel.ManagementStateFilter_available,"")
+
+			case xnametypes.Chassis:
+				//Mt Chassis: PowerState is for rect status
+				var info rf.Chassis
+				err = json.Unmarshal(stsBody, &info)
+				if (err != nil) {
+					glogger.Errorf("ERROR unmarshalling power payload for '%s': %v",
+						rqURL,err)
+					//Power state unknown, but got a response, so mgmt state OK
+					updateHWState(xname,pcsmodel.PowerStateFilter_Undefined,
+						pcsmodel.ManagementStateFilter_available,
+						"Unable to unmarshal power payload")
+					break
+				}
+				powerState,_ = pcsmodel.ToPowerStateFilter(info.PowerState)
+				updateHWState(xname,powerState,
+					pcsmodel.ManagementStateFilter_available,"")
+
+			case xnametypes.CabinetPDUPowerConnector:
+				var info rf.Outlet
+				err = json.Unmarshal(stsBody, &info)
+				if (err != nil) {
+					glogger.Errorf("ERROR unmarshalling power payload for '%s': %v",
+						rqURL,err)
+					//Power state unknown, but got a response, so mgmt state OK
+					updateHWState(xname,pcsmodel.PowerStateFilter_Undefined,
+						pcsmodel.ManagementStateFilter_available,
+						"Unable to unmarshal power payload")
+					break
+				}
+				powerState,_ = pcsmodel.ToPowerStateFilter(info.PowerState)
+				updateHWState(xname,powerState,pcsmodel.ManagementStateFilter_available,"")
+			default:
+				glogger.Errorf("Error: %s: unknown component type.", ctype)
+				updateHWState(xname,pcsmodel.PowerStateFilter_Undefined,
+					pcsmodel.ManagementStateFilter_available,
+					"Unknown component type")
+		}
+	}
+
+	return nil
+}
+
+// Goroutine for monitoring HW and updating the HW state database.  For now,
+// we'll do all of the checking in a single instance rather than sharding.
+// Since we are using TRS that should be OK -- small systems thread off the
+// work, large systems will use remote/worker mode which will shard things for
+// us.
+
+func monitorHW() {
```

Comment:
I have a concern polling for power state every 5 seconds which is the default as far as I can tell. This will cause problems with Gigabyte hardware during power operations. Unfortunately there is no magic time we can use that would be useful. We have been told to wait at least 30 seconds and up to 2 minutes before polling the GB BMC after a power operation has been issued.

Reply:
The sample interval is set in the PowerStatusMonitorInit() function.  It must be set or it's
an error.   An upcoming push will default to 30 seconds, and it will be settable via env var.

Reply:
For GB nodes 30 seconds may still be too fast but otherwise OK.